test/reduce.js

```diff
@@ -13,6 +14,16 @@ describe('reduce', function() {
         assert.strictEqual(R.reduce(mult, 1, [1, 2, 3, 4]), 24);
     });
 
+    it('folds functions over iterables with the supplied accumulator', function() {
+        assert.strictEqual(R.reduce(add, 0, iterdone.range(1, 5)), 10);
+        assert.strictEqual(R.reduce(mult, 1, iterdone.range(1, 5)), 24);
+    });
+
+    it('folds functions over duck typed iterators with the supplied accumulator', function() {
+        assert.strictEqual(R.reduce(add, 0, iterdone.range(1, 5)[iterdone.symbol]()), 10);
+        assert.strictEqual(R.reduce(mult, 1, iterdone.range(1, 5)[iterdone.symbol]()), 24);
+    });
+
```

Comment:
needs a test with logically equivalent objects and non-commutative function, but results should be equal.

otherwise i like this -- more general than my `foldObj`, should have consistent behavior.